src/parser/esprima_cpp/esprima.cpp

```diff
@@ -3496,11 +3496,11 @@ class Parser : public gc {
         RefPtr<Node> exprNode;
         ScanExpressionResult expr;
 
-        if (!this->context->allowYield && this->matchKeyword(YieldKeyword)) {
+        if ((!this->context->allowYield == true || this->config.parseSingleFunction == true) && this->matchKeyword(YieldKeyword)) {
```

Comment:
For readability / State expression explicitly
if ((this->context->allowYield == false || this->config.parseSingleFunction == true) && this->matchKeyword(YieldKeyword) == true)